executor/aggregate.go

```diff
@@ -331,22 +343,23 @@ func recoveryHashAgg(output chan *AfFinalResult, r interface{}) {
 	logutil.BgLogger().Error("parallel hash aggregation panicked", zap.Error(err))
 }
 
-func (w *HashAggPartialWorker) run(ctx sessionctx.Context, waitGroup *sync.WaitGroup, finalConcurrency int) {
-	needShuffle, sc := false, ctx.GetSessionVars().StmtCtx
+func (w *HashAggPartialWorker) run(e *HashAggExec, waitGroup *sync.WaitGroup, finalConcurrency int) {
```

Comment:
Can we add a memTraker attribute for HashAggPartialWorker instead of passing HashAggExec as an input argument? For my part, it's a little wired.

Reply:
sounds good, I'll just add a pointer to the memtracker in the executor in each worker.